frontend/components/idioms.jsx

```diff
@@ -52,91 +52,72 @@ export function UseEffect({ fetchURL, label }) {
 }
 
 export function UseEffectDerivedCalculation() {
-  const [remainder, setReminder] = useState()
-  const [clickedTimes, setClickedTimes] = useState()
-
-  useEffect(() => {
-    setReminder(clickedTimes % 5)
-  }, [clickedTimes])
-
-  const handleClick = () => setClickedTimes(clickedTimes + 1)
+  const [clickedTimes, setClickedTimes] = useState(0); // default state
 
+  const handleClick = () => setClickedTimes(clickedTimes + 1);
   return (
     <div>
       <button onClick={handleClick}>Add Click Count</button>
-      <span>
-        {sum}
-      </span>
-      <span>
-        {remainder}
-      </span>
+      <span>{clickedTimes}</span>
+      <span>{clickedTimes % 5}</span>
```

Comment:
This is better than the orignial solution, but it is also not ideal to put calculations inline like this. It should be a const reminder = clickedTimes % 5 before the jsx.